shared/cache.go

```diff
@@ -165,23 +167,36 @@ func (mc memcacheReadWritable) NewReadCloser(iKey interface{}) (io.ReadCloser, e
 	if !ok {
 		return nil, errNewReadCloserExpectedString
 	}
+	if Clients.redisPool == nil {
+		return nil, errNoRedis
+	}
+	conn := Clients.redisPool.Get()
+	defer conn.Close()
 
-	item, err := memcache.Get(mc.ctx, key)
-	if err == memcache.ErrCacheMiss {
-		return nil, errCacheMiss
-	} else if err != nil {
+	// https://redis.io/commands/get
+	result, err := conn.Do("GET", key)
+	if err != nil {
 		return nil, err
+	} else if result == nil {
+		return nil, errCacheMiss
 	}
-	return ioutil.NopCloser(bytes.NewReader(item.Value)), nil
+	b, ok := result.([]byte)
+	if !ok {
+		return nil, errMemcacheInvalidResponseType
+	}
+	return ioutil.NopCloser(bytes.NewReader(b)), nil
 }
 
 func (mc memcacheReadWritable) NewWriteCloser(iKey interface{}) (io.WriteCloser, error) {
 	key, ok := iKey.(string)
 	if !ok {
 		return nil, errNewReadCloserExpectedString
 	}
-
-	return &memcacheWriteCloser{mc, key, mc.expiry, bytes.Buffer{}, false, false}, nil
+	var conn redis.Conn
+	if Clients.redisPool != nil {
+		conn = Clients.redisPool.Get()
```

Comment:
I don't see any immediate benefit between creating `conn` here vs in `Close()` but if I have to choose, I will put it in `Close()`. It allows `conn` to be created and closed immediately

Reply:
Yeah I agree. I was going to write to Redis in `Write` but decided not to eventually. Let me fix it up.